fix: ensure terminal windows don't part from parent window when toggling visibility

AlexJuca opened the pull request:
As of version 1.0.1 (macOS build) when running the toggle visibility action a window with tabs is made into multiple windows. 

This PR ensures terminal tabs are reconstructed and correctly placed into its parent window.

# Demo

https://github.com/user-attachments/assets/44f14bca-15a1-4717-ba0a-44f0767feec3


FYI: I will create another PR to ensure the right tab is focused after the main window is restored.

Fixes #4329

AlexJuca committed:
fix: ensure terminal tabs are reconstructed in main window after toggling visibility

mitchellh reviewed (commented):
I haven't tested this but the source looks a bit suspicious to me. Does this force all windows back into the same main window? Or does it reconstruct multiple tabbed windows properly? If its the former then we need to go back to the drawing board on this one...

AlexJuca commented:
@mitchellh You're right. Let me take another jab at this one.

mitchellh commented:
Thanks. Closing now

AlexJuca commented:
> Thanks. Closing now

@mitchellh Actually, you do not need to close this one as I intend to use the same PR for the new approach. Is that fine?

mitchellh commented:
I'd prefer you open a new PR when ready otherwise it's unclear what is sitting and what is WIP. I can perhaps convert it to a draft though let me see.

mitchellh commented:
I don't think I can convert it, but perhaps you can convert it to a draft? Link should be under "Reviewers"

AlexJuca committed:
fix: hide windows without calling orderOut API

AlexJuca committed:
chore: revert on TerminalManager changes

AlexJuca commented:
@mitchellh A side effect of .orderOut() approach is that child windows are removed from their NSTabWindowGroup when hidden, and calling .addTabbedWindow() when unhiding does not successfully re-associate them with their parent group. This appears to be a limitation of how NSTabWindowGroup manages windows in macOS.

I've implemented a new approach, which simplifies the process of hiding and showing windows and avoids the use of the .orderOut() function. This approach ensures windows remain part of the macOS window hierarchy.

The current solution uses alpha value manipulation (alphaValue = 0 to hide and alphaValue = 1 to show) for all windows. 
This ensures Windows remain part of their original NSTabGroup: The alpha value approach keeps the windows logically attached to their parent group, avoiding group fragmentation.

	
Here is a quick demo:

https://github.com/user-attachments/assets/b62a6c5b-3a77-4c65-adfb-dc0ca213b023

Let me know what you think of this approach.

mitchellh committed:
macos: rework toggle_visibility to better match iTerm2

Two major changes:

1. Hiding uses `NSApp.hide` which hides all windows, preserves tabs, and
   yields focus to the next app.

2. Unhiding manually tracks and brings forward only the windows we hid.
   Proper focus should be retained.

mitchellh commented:
Thanks! Your work worked but then I realized that `toggle_visibility` has a bunch of weird behaviors that didn't match iTerm2. I reworked that now and fixed those issues. See my commit. Thanks for your help!
